theme/js/components/version-selector.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const $ = global.jQuery;
+
+class VersionSelector {
+  constructor() {
+    this._getVersions(versions => {
+      this._render(versions);
+    });
+  }
+
+  get currentComponent() {
+    return window.location.pathname.split('/').slice(2).join('/') || null;
+  }
+
+  get currentVersion() {
+    return window.location.pathname.split('/')[1] || '';
+  }
+
+  toggleSelector() {
+    const button = $('#version_selector_button');
+    const dropdownMenu = $('#version_selector_dropdown_menu');
+    if (button.attr('aria-expanded') === 'true') {
+      dropdownMenu.hide();
+      button.attr('aria-expanded', 'false');
+    }
+    else {
+      dropdownMenu.show();
+      button.attr('aria-expanded', 'true');
+    }
+  }
+
+  _getVersions(callback) {
+    const versionRequest = new XMLHttpRequest();
```

Comment:
Nu we een bundler en module loader hebben kunnen we er ook voor kiezen om iets als `axios` te installeren en een `VersionService` te maken. Deze service zou dan ook in de global scope beschikbaar kunnen zijn zodat de versions pagina (https://dso-toolkit.nl/11.1.0/docs/versions.html) daar ook gebruik van maakt.. En die kan het resultaat cachen waardoor er maar één request naar versions.json wordt gedaan.

Wat denk je?

Reply:
Doen we in ander issue